proposals/records-wip.md

```diff
@@ -28,3 +28,35 @@ struct_body
 ```
 
 The `attributes` non-terminal will also permit a new contextual attribute, `data`.
+
+A class (struct) declared with a parameter list or `data` modifier is called a record class (record struct), either of which is a record type.
+
+It is an error to declare a record type without both a parameter list and the `data` modifier.
+
+## Members of a record type
+
+In addition to the members declared in the class-body, a record type has the following additional members:
+
+### Primary Constructor
+
+A record type has a public constructor whose signature corresponds to the value parameters of the
+type declaration. This is called the primary constructor for the type, and causes the implicitly
+declared default constructor to be suppressed. If a constructor with the same signature is
+already present in the class, the synthesized primary constructor is suppressed.
+At runtime the primary constructor 
+
+1. initializes compiler-generated backing fields for the properties corresponding to the value parameters (if these properties are compiler-provided; see [Synthesized properties](#Synthesized Properties)); then
+
+2. executes the instance field initializers appearing in the class-body; and then
+    invokes the base class constructor with no arguments.
+
+[ ] TODO: add base call syntax and specification about choosing base constructor through overload resolution
+
+### Properties
+
+For each record parameter of a record type declaration there is a corresponding public property member whose name and type are taken from the value parameter declaration. If no concrete (i.e. non-abstract) public property with a get accessor and with this name and type is explicitly declared or inherited, it is produced by the compiler as follows:
```

Comment:
>public [](start = 206, length = 6)

probably delete `public`.  I think you want to permit an explicit declaration to change the protection.

Reply:
The naming convention for parameters is lower case, and for properties is uppercase. It's there any intention of fixing up the casing?

Reply:
Our current thinking is "no".  Just use uppercase.

---
In reply to: [364061234](https://github.com/dotnet/csharplang/pull/3076#discussion_r364061234) [](ancestors = 364061234)